### ISSUE TITLE
when optimizer is sgd, the loss will be NaN

### ISSUE THREAD

**Issue by cheyennee** [keras-team/keras#18919]
I have a keras [model](https://drive.google.com/file/d/1gPdHdTqARXBNoXRRcG75MBg4Rd7rKBsN/view?usp=sharing) here. When I use sgd as the optimizer of this model, the loss value will be NaN. However, when I use other optimizer, such as rmsprop, the loss outputs normally. I'm not sure if there is a bug in sgd.

Repo snippet code is in [colab](https://colab.research.google.com/drive/1LzTHNud-sIK6YRSrkCptF-Sm3SAGhuuz?usp=sharing). Need to download this keras model first.

**Comment by Frightera** [comment]
SGD is not an adaptive optimizer, so the results are expected. What you are seeing is called "Gradient Explosion", in a nutshell SGD can't deal with big input numbers unless you clip the gradients etc.

**Comment by sachinprasadhs** [comment]
@Frightera , Thanks for your input.
@cheyennee , as mentioned in the above comment, please use the optimizers from the list here based on your model design https://keras.io/api/optimizers/

**Comment by cheyennee** [comment]
Thanks, @Frightera @sachinprasadhs .I will change other optimizers.
